src/axios-har-tracker.ts

```diff
@@ -1,112 +1,168 @@
 
-async function axiosTracker(call) {
-  axios.interceptors.request.use(
-    async config => {
-      config.validateStatus = function () {
-        return true;
-      };
-      config.headers['request-startTime'] = process.hrtime();
-      const fullCookie = JSON.stringify(config.headers['Cookie']);
-      const version = config.httpVersion === undefined ? 'HTTP/1.1' : 'HTTP/' + config.httpVersion;
-
-      newEntry.request = {
-        method: config.method,
-        url: config.url,
-        httpVersion: version,
-        cookies: getCookies(fullCookie),
-        headers: [],
-        queryString: getParams(config.params),
-        headersSize: -1,
-        bodySize: -1
-      };
-      return config;
-    },
-    error => {
-      return Promise.reject(error);
-    }
-  );
-
-  axios.interceptors.response.use(
-    async resp => {
-      if (resp) {
-        newEntry.response = {
-          status: resp.status,
-          statusText: resp.statusText,
-          headers: getHeaders(resp.headers),
-          startedDateTime: new Date(resp.headers.date),
-          time: resp.headers['request-duration'] = Math.round(
-            process.hrtime(resp.headers['request-startTime'])[0] * 1000 +
-              process.hrtime(resp.headers['request-startTime'])[1] / 1000000
-          ),
-          httpVersion: `HTTP/${resp.request.res.httpVersion}`,
-          cookies: getCookies(JSON.stringify(resp.config.headers['Cookie'])),
-          bodySize: JSON.stringify(resp.data).length,
-          redirectURL: '',
-          headersSize: -1,
-          content: {
-            size: JSON.stringify(resp.data).length,
-            mimeType: resp.headers['content-type'] ? resp.headers['content-type'] : 'text/plain',
-            text: JSON.stringify(resp.data)
-          },
-          cache: {},
-          timings: {
-            blocked: -1,
-            dns: -1,
-            ssl: -1,
-            connect: -1,
-            send: 10,
-            wait: 10,
-            receive: 10,
-            _blocked_queueing: -1
-          }
-        };
-        const enteriesContent = Object.assign({}, newEntry);
-        generatedHar.log.entries.push(enteriesContent);
+export class AxiosHarTracker {
+
+  private axios: AxiosStatic;
+  private generatedHar: HarFile;
+  private date: string;
+  private newEntry: NewEntry;
+
+  constructor(axiosModule: AxiosStatic) {
+    this.axios = axiosModule;
+    this.generatedHar = {
+      log: {
+        version: '1.2',
+        creator: {
+          name: 'axios-har-tracker',
+          version: '0.1.0'
+        },
+        pages: [],
+        entries: []
+      }
+    };
+    this.date = new Date().toISOString();
+
+    this.axios.interceptors.request.use(
+
+      async config => {
+        this.newEntry = this.generateNewEntry();
+        this.newEntry.request = this.returnRequestObject(config);
+        return config;
+      },
+      async error => {
+        this.newEntry.request = this.returnRequestObject(error.request);
```

Comment:
Add `if(error.request)` because error object not always will contain request property. Look here https://github.com/axios/axios#handling-errors